### PR TITLE
Test/e2e/alert

### PR DESCRIPTION
**Pull request by agonzalez-r**
# Alert E2E

## Description
This PR introduces an end-to-end test for the flow of creating an alert and editing it and deleting it. 

## Changes
- Added the end-to-end test 
- Changed test tags in `AlertLists.kt`, because using the alert.id in the test tag was not possible from the end to end. 

#### Added
- app/src/androidTest/java/com/android/periodpals/endtoend/EndToEndAlert.kt

#### Modified
- `AlertLists.kt`
- `AlertListsScreenTest`
#### Removed
None.

## Dependencies Added
None.

### CODE REVIEW

File: app/src/main/java/com/android/periodpals/ui/alert/AlertLists.kt [hunk 1]
```diff
@@ -292,9 +293,10 @@ fun AlertListsScreen(
             if (myAlertsList.isEmpty()) {
               item { NoAlertDialog(context.getString(R.string.alert_lists_no_my_alerts_dialog)) }
             } else {
-              items(myAlertsList) { alert ->
+              itemsIndexed(myAlertsList) { index, alert ->
```
Comment (charliemangano) [inline app/src/main/java/com/android/periodpals/ui/alert/AlertLists.kt:296]: Good fix!

File: app/src/main/java/com/android/periodpals/ui/alert/AlertLists.kt [hunk 2]
```diff
@@ -352,6 +356,7 @@ fun AlertListsScreen(
  * profile picture, time, location, product type, urgency, and an edit button.
  *
  * @param alert The alert to be displayed.
+ * @param indexTestTag The index of the alert in the list.
```
Comment (charliemangano) [inline app/src/main/java/com/android/periodpals/ui/alert/AlertLists.kt:359]: Nice reflex

**Review by charliemangano — APPROVED** [review]
Good implementation of the end-to-end tests! You found a way to make the test tags work, good job!

**Review by coaguila — APPROVED** [review]
## Summary
Great end to end test for Alerts!
## Important (which includes)
#### Code Quality
LGTM
#### Functionality
LGTM
## Testing
Very cohesive tests should correctly reflect users' flow.

**Pull request merged**